src/index.tsx

```diff
@@ -233,13 +278,14 @@ class PyflyByWidget extends Widget {
             const currentLockId = this._lock.requestedLockCount;
             this._lock.enable(currentLockId + 1);
             await this._lock.promise[currentLockId];
-            this._sendFormatCodeMsg(imports);
+            this._sendFormatCodeMsg(imports, currentLockId);
           });
           break;
         }
         case PYFLYBY_COMMS.FORMAT_IMPORTS: {
           this._formatImports(msgContent);
-          this._lock.disable();
+          const { lock_id: lockId }: any = msgContent;
+          this._lock.disable(lockId + 1);
```

Comment:
Why disable id + 1?

Reply:
I see this wasn't very intuitive, I have refactored the code to make this slightly easier to understand.